recipe/activate-go-cgo.bat

```diff
@@ -0,0 +1,21 @@
+@echo off
+set "CONDA_BACKUP_GOROOT=%GOROOT%"
+set "GOROOT=%CONDA_PREFIX%\go"
+
+set "CONDA_BACKUP_CGO_ENABLED=%CGO_ENABLED%"
+set "CGO_ENABLED=1"
+
+set "CONDA_BACKUP_CGO_CFLAGS=%CGO_CFLAGS%"
+set "CGO_CFLAGS=%CFLAGS%"
+
+set "CONDA_BACKUP_CGO_CPPFLAGS=%CGO_CPPFLAGS%"
+set "CGO_CPPFLAGS=%CPPFLAGS%"
+
+set "CONDA_BACKUP_CGO_CXXFLAGS=%CGO_CXXFLAGS%"
+set "CGO_CXXFLAGS=%CXXFLAGS%"
+
+set "CONDA_BACKUP_CGO_FFLAGS=%CGO_FFLAGS%"
+set "CGO_FFLAGS=%FFLAGS%"
+
+set "CONDA_BACKUP_CGO_LDFLAGS=%CGO_LDFLAGS%"
+set "CGO_LDFLAGS=%LDFLAGS%"
```

Comment:
At the moment the `m2w64_` compilers do not set any flags, but if that gets fixed the `go` recipe would use flags by default.